sentry-android-core/src/main/java/io/sentry/android/core/SentryPerformanceProvider.java

```diff
@@ -204,8 +205,36 @@ private void onAppLaunched(
 
     activityCallback =
         new ActivityLifecycleCallbacksAdapter() {
+
+          @Override
+          public void onActivityCreated(
+              @NotNull Activity activity, @Nullable Bundle savedInstanceState) {
+            // In case the SDK gets initialized async or the
+            // ActivityLifecycleIntegration is not enabled (e.g on RN due to Context not being
+            // instanceof Application)
+            // the app start type never gets set
+            if (appStartMetrics.getAppStartType() == AppStartMetrics.AppStartType.UNKNOWN) {
+              // We consider pre-loaded application loads as warm starts
+              // This usually happens e.g. due to BroadcastReceivers triggering
+              // Application.onCreate only, but no Activity.onCreate
+              final long now = SystemClock.uptimeMillis();
+              final long durationMs =
+                  now - appStartMetrics.getAppStartTimeSpan().getStartUptimeMs();
+              if (durationMs > TimeUnit.SECONDS.toMillis(1)) {
```

Comment:
If I understood it right, we had this condition previously before https://github.com/getsentry/sentry-java/pull/3937, but somewhere else?

If we had it, it seems safe, but if not, it might be quite dangerous as the difference between Flutter Debug and Release showed.

Reply:
Yes, it's within AppStartMetrics. I've now [extracted the parts ](https://github.com/getsentry/sentry-java/pull/4216/files#diff-459be72344729074d45910e2102dbbb5dc1878f9838060380f95c578cbabda12R363-R378) so they can be re-used. And there was a bug in there as well: Using 1 second instead of 1 minute as a threshold 🙈